apollo/lib/websocket/app.py

```diff
@@ -0,0 +1,74 @@
+import uuid
+import json
+from typing import Dict, Set
+
+from fastapi import WebSocket
+from pydantic.json import pydantic_encoder
+from starlette.websockets import WebSocketDisconnect
+
+from apollo.lib.singleton import Singleton
+from apollo.lib.websocket import ConnectionManager
+from apollo.lib.websocket.interest_type import WebSocketObserverInterestType
+
+
+class AppConnectionManager(ConnectionManager, metaclass=Singleton):
+    interested_connections: Dict[WebSocketObserverInterestType,
+                                 Set[uuid.UUID]] = {}
+
+    @staticmethod
+    async def _listen(connection: WebSocket):
+        try:
+            while True:
+                await connection.receive_text()
+        except WebSocketDisconnect:
+            return
+
+    def _add_interested_connection(
+        self,
+        observer_interest_type: WebSocketObserverInterestType,
+        connection_id: uuid.UUID
+    ):
+        connections = self.interested_connections.get(observer_interest_type,
+                                                      set())
+        connections.add(connection_id)
+        self.interested_connections[observer_interest_type] = connections
+
+    async def _close_and_remove_connection(self, connection_id: uuid.UUID):
+        await self.close_connection(connection_id)
+        for connection_list in self.interested_connections.values():
+            try:
+                connection_list.remove(connection_id)
+            except ValueError:
```

Comment:
`Set.remove(elem)` raises a [KeyError](https://docs.python.org/3.8/library/stdtypes.html#frozenset.remove) if the element isn't present. However, I'd use `Set.discard(elem)` which doesn't raise anything.